### PR TITLE
Dividing Tasks to Student in C++

### LINKED ISSUE #5456: Dividing Tasks to student [CP][GSSOC'21]
## 💥 Proposal

Dividing Tasks to student is CP problem that state that  There are N students in the class, i'th student has Ci task in his collection. Your task is to determine if it is possible to redistribute C1 + C2 + ... + Cn task among the student of class thus that i'th student would get i task.

[GSSOC'21][CP][C-Plus-Plus]

### Have you read the [Contributing Guidelines on Pull Requests][ ](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
Yes
(Write your answer here.)

### PR DESCRIPTION
**Pull request by vedant-jain03**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

* Yes

### Description

The Problem states that Given array arr[n] of integer where each element arr[i] represent number of task given to each student.Your task is to find if it possible to redistribute the task where i'th student assign i number of task. Example is it possible to assign 2 task to 2nd Student. 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

fixes #5456

### CODE REVIEW

**Comment by vedant-jain03** [comment]
@HarshCasper 
I don't understand why checks are wrong. Please review and give me feedback

**Comment by rohanbabbar04** [comment]
@vedant-jain03 make your branch even with the main branch of tesseract coding and then commit
But still, it can get merged so you don't need to worry

File: C-Plus-Plus/cp/Dividing_tasks.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+The Problem states that Given array arr[n] of integer where each element arr[i] represent number of task given to each student.Your task is to find if it possible to redistribute the task where i'th student assign i number of task. Example is it possible to assign 2 task to 2nd Student. 
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Dividing_tasks.cpp]: Rewrite it like that

```
/*
The Problem states that Given array arr[n] of integer where each element arr[i] represent 
number of task given to each student.Your task is to find if it possible to redistribute the 
task where i'th student assign i number of task. Example is it possible to assign 2 task to 2nd Student. 
*/
```

File: C-Plus-Plus/cp/Dividing_tasks.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#include <bits/stdc++.h>
+using namespace std;
+void solve(int arr[], int n)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Dividing_tasks.cpp:9]: Add a new line after std;

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
- Add sample I/O at the end of the Code with proper display message using cout
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

File: C-Plus-Plus/cp/Dividing_tasks.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+int main()
+{
+    int n;
+    cin >> n;
+    int arr[n];
+    for (int i = 1; i <= n; i++)
+    {
+        cin >> arr[i];
+    }
+    solve(arr, n);
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Dividing_tasks.cpp]: Use proper display message using `cout` while reading and writing the input and output

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Do the changes as mentioned.

**Review by vedant-jain03 — COMMENTED** [review]
Done

File: C-Plus-Plus/cp/Dividing_tasks.cpp (hunk 1 above)
Comment (ankitaggarwal23) [reply C-Plus-Plus/cp/Dividing_tasks.cpp]: Resolve this issue

**Comment by ankitaggarwal23** [comment]
> ### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
> * Yes
> 
> ### Description
> The Problem states that Given array arr[n] of integer where each element arr[i] represent number of task given to each student.Your task is to find if it possible to redistribute the task where i'th student assign i number of task. Example is it possible to assign 2 task to 2nd Student.
> 
> ### Checklist
> * [x]  I've read the contribution guidelines.
> * [x]  I've checked the issue list before deciding what to submit.
> * [x]  I've edited the `README.md` and link to my code.
> 
> ## Related Issues or Pull Requests
> fixes #5456

Do the mentioned changes

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Do the mentioned changes still one is left

File: C-Plus-Plus/cp/Dividing_tasks.cpp (hunk 1 above)
Comment (vedant-jain03) [reply C-Plus-Plus/cp/Dividing_tasks.cpp]: "The problem state that given n number of students in the class and for i'th student given arr[i] represent number of task assigned to that student. Teacher who assigned task to them is getting confuse with the numbers. So she decided to redistribute task in such a way that ,i'th student will get i number of task ( example, 2nd student will get 2 tasks).
Your task is to find if it is possible to redistribute the task. Help the teacher!"

@ankitaggarwal23 Is this sound good mentor or any changes

Comment (ankitaggarwal23) [reply C-Plus-Plus/cp/Dividing_tasks.cpp]: No just change it as mentioned rest looks good

**Review by vedant-jain03 — COMMENTED** [review]
Done @ankitaggarwal23

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by vedant-jain03** [comment]
@ankitaggarwal23 Thankyou for merging PR 
please review my this PR also
#5450

**Pull request merged**